source/slang/slang-emit-c-like.cpp

```diff
@@ -3513,6 +3539,13 @@ void CLikeSourceEmitter::emitFuncDecorationsImpl(IRFunc* func)
 
 void CLikeSourceEmitter::emitStruct(IRStructType* structType)
 {
+    // TODO: add similar functionality like markTypeForPyExport for [HeaderExport]
+
+    // When emmiting header, skip if should not be emmited.
+    if (isEmmitingHeader() && !structType->findDecoration<IRHeaderExportDecoration>()) {
```

Comment:
There may be an issue here if an exported struct has a member whose type is not exported.